feat(survey): Enable surveys to be repeated across a time period.

Phanatic opened the pull request:
## Problem

This PR enables survey creators to setup a recurring schedule for their surveys to be shown with distinct 
iterations and the ability to track NPS scores across the timeline.

Related issue : https://github.com/PostHog/posthog/issues/22099

### Definition 
1. - [x] Allow survey creators to define a repeatable schedule for surveys
 ![image](https://github.com/PostHog/posthog/assets/82819/add040b1-0acb-465f-9ab7-407876b23f79)
3. - [x] Do not allow removing options or questions if a survey is repeatable.
4. - [x] Update current_iteration and survey dates based on today’s date.
5. - [x] Add a payment gate for this paid feature.

### Collection 
1. - [x] Send back survey iteration and start date in survey responses. 
2. - [x] Do not show survey within the same iteration to a user. 

### Reflection
1. - [x] Show overall NPS
2. - [x] Show current iteration’s NPS
3. - [x] Show upcoming iteration, ability to increase or decrease iterations.
4. - [x] Show a chart of NPS values from different iterations
![image](https://github.com/PostHog/posthog/assets/82819/9cf8679d-f9da-4911-9b4b-c76cbb0aaf7c)
6. - [ ] Add storyboards for new views
 

## Changes

<!-- If there are frontend changes, please include screenshots. -->
<!-- If a reference design was involved, include a link to the relevant Figma frame! -->

👉 _Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review._

## Does this work well for both Cloud and self-hosted?

<!-- Yes / no / it doesn't have an impact. -->

## How did you test this code?

<!-- Briefly describe the steps you took. -->
<!-- Include automated tests if possible, otherwise describe the manual testing routine. -->

Commit:
saving migrations

Commit:
saving work

Commit:
Can round-trip iteration_count and iteration_repeat_days

Commit:
Can roundtrip iteration_frequency

Commit:
Can see repeated survey occurences in view

Commit:
save some fixes

Commit:
Merge branch 'master' into repetable-surveys

Commit:
fix migrations

Commit:
update taxonomy

Commit:
Merge branch 'master' into repetable-surveys

Commit:
update migrations

Commit:
Can show NPS scores over time

Commit:
remove unnecessary changes

Commit:
remove unnecessary changes

Commit:
Merge branch 'master' into repetable-surveys

Commit:
rename migration file

Commit:
Add type annotations to task

github-actions[bot] committed:
Update UI snapshots for `chromium` (2)

Commit:
Address mypy concerns

Commit:
fix tsc errors

Phanatic committed:
Merge branch 'master' into repetable-surveys

Commit:
Add tests

Commit:
Merge branch 'master' into repetable-surveys

Commit:
Add more tests for creating surveys with recurring schedules

github-actions[bot] committed:
Update UI snapshots for `chromium` (1)

github-actions[bot] committed:
Update UI snapshots for `chromium` (1)

Phanatic committed:
Merge branch 'master' into repetable-surveys

Commit:
fix visuals for scheduling surveys

Commit:
Merge branch 'master' into repetable-surveys

Commit:
update migrations

Phanatic committed:
Merge branch 'master' into repetable-surveys

github-actions[bot] committed:
Update UI snapshots for `chromium` (2)

Commit:
only show recurring NPS Scores until current iteration

Phanatic committed:
Merge branch 'master' into repetable-surveys

Commit:
Add tests to ensure that internal targeting flag is updated

Phanatic committed:
Merge branch 'master' into repetable-surveys

Commit:
Fix python codelint errors

Commit:
fix tests

Phanatic committed:
Merge branch 'master' into repetable-surveys

github-actions[bot] committed:
Update query snapshots

Commit:
Guard recurring surveys for paid features

github-actions[bot] committed:
Update UI snapshots for `chromium` (2)

github-actions[bot] committed:
Update UI snapshots for `chromium` (2)

github-actions[bot] committed:
Update UI snapshots for `chromium` (2)

github-actions[bot] committed:
Update query snapshots

github-actions[bot] committed:
Update UI snapshots for `chromium` (2)

github-actions[bot] committed:
Update query snapshots

github-actions[bot] committed:
Update UI snapshots for `chromium` (2)

Commit:
fix typecheck error

Commit:
Merge branch 'master' into repetable-surveys

neilkakkar commented on posthog/tasks/update_survey_iteration.py:
```diff
@@ -0,0 +1,87 @@
+
+def update_survey_iteration() -> None:
+    surveys_with_recurring_schedules = (
+        Survey.objects.exclude(Q(end_date__isnull=False))
+        .exclude(Q(iteration_count=0), Q(iteration_count=F("current_iteration")))
```
I think we also need to include here surveys that have `iteration_count` set? As otherwise they're not really recurring surveys.

I'm not sure what query the F expression generates (can you add a @snapshot_postgres_queries decorator to test above for update_survey_iteration?) but I think we can just rewrite this to be:

neilkakkar replied:
```suggestion
        Survey.objects.filter(end_date__isnull=True, iteration_count__isnull=False))
```

neilkakkar reviewed (commented):
Great work, this is a big chonky feature :) - some comments

Phanatic commented on frontend/src/scenes/surveys/SurveyEdit.tsx:
```diff
@@ -645,6 +659,98 @@ export default function SurveyEdit(): JSX.Element {
+                                        {({ onChange }) => {
+                                            return (
+                                                <LemonRadio
+                                                    value={schedule}
```
ahh, that's what I was looking for, good callout, fixed in my next commit

Phanatic replied:
fixed!

Commit:
use selector to manage showing recurring schedule input

Phanatic commented on frontend/src/scenes/surveys/SurveyEdit.tsx:
```diff
@@ -645,6 +659,98 @@ export default function SurveyEdit(): JSX.Element {
+                                                    onChange={(newValue) => {
+                                                        onChange(newValue)
+                                                        if (newValue === 'once') {
+                                                            setShowSurveyRepeatSchedule(false)
```
good callout, fixing now.

Phanatic replied:
fixed!

Phanatic commented on frontend/src/scenes/surveys/SurveyEdit.tsx:
```diff
@@ -645,6 +659,98 @@ export default function SurveyEdit(): JSX.Element {
+                                                            onChange={(newValue) => {
+                                                                if (newValue && newValue > 0) {
+                                                                    onChange(newValue)
+                                                                    setSurveyValue('iteration_count', newValue)
```
🔥

Phanatic commented on frontend/src/scenes/surveys/SurveyView.tsx:
```diff
@@ -305,17 +322,34 @@ export function SurveyResult({ disableEventsTable }: { disableEventsTable?: bool
                                     surveyRatingResults={surveyRatingResults}
                                     surveyRatingResultsReady={surveyRatingResultsReady}
                                     questionIndex={i}
+                                    iteration={survey.current_iteration}
```
This is something I'm getting feedback today on, with the customer who filed this bug. 
We have a way to get the NPS results for the current iteration right above this chart, and we retain the "current iteration" view here. 
I dunno if it makes more sense to show all responses since the start of the survey here, or just the ones received in the current iteration. I'll make a call later today.

neilkakkar replied:
I meant this specifically for non-NPS rating charts, where we have no extra calculations like we do for NPS.

Phanatic commented on frontend/src/types.ts:
```diff
@@ -86,6 +86,7 @@ export enum AvailableFeature {
     SURVEYS_LINK_QUESTION_TYPE = 'surveys_link_question_type',
     SURVEYS_SLACK_NOTIFICATIONS = 'surveys_slack_notifications',
     SURVEYS_WAIT_PERIODS = 'surveys_wait_periods',
+    SURVEYS_RECURRING = 'surveys_recurring',
```
Billing updates are in this PR https://github.com/PostHog/billing/pull/650

Phanatic commented on posthog/api/survey.py:
```diff
@@ -255,7 +264,32 @@ def update(self, instance: Survey, validated_data):
+        instance.iteration_frequency_days = validated_data.get("iteration_frequency_days")
+
         instance = super().update(instance, validated_data)
+        # if "iteration_count" in validated_data and "iteration_frequency_days" in validated_data:
```
I moved all this code to the model and removed this from here.

Phanatic commented on posthog/api/survey.py:
```diff
@@ -255,7 +264,32 @@ def update(self, instance: Survey, validated_data):
+        #     instance.current_iteration_start_date = None
+        #     instance.save()
+
         self._add_user_survey_interacted_filters(instance, end_date)
```
good eye, this will need to take into account the iteration number too, fixing now.

Phanatic commented on posthog/tasks/update_survey_iteration.py:
```diff
@@ -0,0 +1,87 @@
+    survey.current_iteration = _get_current_iteration(survey)
+    survey.current_iteration_start_date = survey.iteration_start_dates[survey.current_iteration - 1]
+    survey.save(update_fields=["current_iteration", "current_iteration_start_date"])
+    _update_internal_targeting_flag(survey)
```
you're right, we can reduce this to only call save when necessary and call save only once on the Survey object here.

Commit:
CR feedback

Phanatic commented on frontend/src/scenes/surveys/SurveyEdit.tsx:
```diff
@@ -55,10 +58,21 @@ export default function SurveyEdit(): JSX.Element {
+    const surveysRecurringScheduleDisabledReason = surveysRecurringScheduleAvailable
+        ? undefined
+        : 'Subscribe to surveys for multiple questions'
+    // setSchedule(survey.iteration_frequency_days === 0 ? 'once' : 'recurring')
```
removed now!

Phanatic commented on frontend/src/scenes/surveys/SurveyView.tsx:
```diff
@@ -217,6 +218,20 @@ export function SurveyView({ id }: { id: string }): JSX.Element {
+                                                        <div className="flex flex-col">
+                                                            <span className="card-secondary mt-4">Schedule</span>
+                                                            <span>
+                                                                Repeats every {survey.iteration_frequency_days} days,{' '}
```
I found the `pluralize` function! fixed now.

Phanatic commented on frontend/src/scenes/surveys/surveyLogic.tsx:
```diff
@@ -475,6 +584,8 @@ export const surveyLogic = kea<surveyLogicType>([
             actions.setSurveyValue('conditions', NEW_SURVEY.conditions)
             actions.setSurveyValue('remove_targeting_flag', true)
             actions.setSurveyValue('responses_limit', NEW_SURVEY.responses_limit)
+            actions.setSurveyValue('iteration_count', NEW_SURVEY.iteration_count)
```
nice, fixed now

Phanatic commented on frontend/src/scenes/surveys/surveyViewViz.tsx:
```diff
@@ -247,6 +249,89 @@ export function RatingQuestionBarChart({
+                    <div className="text-xl font-bold mb-2">NPS Scores over time for "{question.question}"</div>
+                    <div className=" h-50 border rounded pt-8">
+                        <div className="relative h-full w-full">
+                            <BindLogic logic={insightLogic} props={insightProps}>
```
Fixed now, made a new Props type

Phanatic committed:
Merge branch 'master' into repetable-surveys

Commit:
Merge branch 'master' into repetable-surveys

github-actions[bot] committed:
Update query snapshots

github-actions[bot] committed:
Update query snapshots

github-actions[bot] committed:
Update UI snapshots for `chromium` (2)

github-actions[bot] committed:
Update query snapshots

Phanatic commented on frontend/src/scenes/surveys/SurveyView.tsx:
```diff
@@ -217,6 +218,20 @@ export function SurveyView({ id }: { id: string }): JSX.Element {
+                                                            <span className="card-secondary mt-4">Schedule</span>
+                                                            <span>
+                                                                Repeats every {survey.iteration_frequency_days} days,{' '}
+                                                                {survey.iteration_count} times
```
fixed with the pluralize changes.

Phanatic commented on frontend/src/scenes/surveys/surveyLogic.tsx:
```diff
@@ -143,6 +157,8 @@ export const surveyLogic = kea<surveyLogicType>([
         setSelectedQuestion: (idx: number | null) => ({ idx }),
         setSelectedSection: (section: SurveyEditSection | null) => ({ section }),
+        setShowSurveyRepeatSchedule: (showSurveyRepeatSchedule: boolean) => ({ showSurveyRepeatSchedule }),
+        setSchedule: (schedule: string) => ({ schedule }),
```
fixed now

Commit:
fix initial show of survey after editing

Commit:
fix typecheck errors

Commit:
Merge branch 'master' into repetable-surveys

github-actions[bot] committed:
Update query snapshots

github-actions[bot] committed:
Update query snapshots

Twixes committed:
chore(api-docs): Fix drf-spectacular warnings causing CI flakiness (#22593)

* chore(api-docs): Fix drf-spectacular warnings causing CI flakiness

* Update query snapshots

---------

Co-authored-by: github-actions <41898282+github-actions[bot]@users.noreply.github.com>

github-actions[bot] committed:
Update query snapshots

github-actions[bot] committed:
Update UI snapshots for `chromium` (1)

github-actions[bot] committed:
Update query snapshots

github-actions[bot] committed:
Update query snapshots

github-actions[bot] committed:
Update UI snapshots for `chromium` (1)

github-actions[bot] committed:
Update UI snapshots for `chromium` (2)

github-actions[bot] committed:
Update query snapshots

github-actions[bot] committed:
Update UI snapshots for `chromium` (2)

github-actions[bot] committed:
Update query snapshots

bretthoerner committed:
fix: add _timestamp filter to event deletion by Person (#22582)

github-actions[bot] committed:
Update query snapshots

github-actions[bot] committed:
Update UI snapshots for `chromium` (2)

mariusandra committed:
feat(hog): f-strings (#22577)

raquelmsmith committed:
feat: unhide the person profiles UI for those who have access (#22595)

Co-authored-by: github-actions <41898282+github-actions[bot]@users.noreply.github.com>

pauldambra committed:
feat: control heatmap settings from in-app heatmap page (#22522)

Co-authored-by: David Newell <d.newell1@outlook.com>
Co-authored-by: github-actions <41898282+github-actions[bot]@users.noreply.github.com>

timgl committed:
chore: Make migrations safe again (#22594)

* chore: Make migrations safe again

* Update query snapshots

* Update query snapshots

---------

Co-authored-by: github-actions <41898282+github-actions[bot]@users.noreply.github.com>
Co-authored-by: James Greenhill <fuziontech@gmail.com>

raquelmsmith committed:
feat: use a flag to define inner or left join orgs (#22561)

Co-authored-by: github-actions <41898282+github-actions[bot]@users.noreply.github.com>

Twixes committed:
fix(insight): Use viewset's `team` in `@cached_by_filters` (#22590)

* fix(insight): Use viewset's `team` in `@cached_by_filters`

* Add test

* Update query snapshots

* Make mypy happy about `DummyViewSet`

* Update query snapshots

* Update query snapshots

* Update query snapshots

---------

Co-authored-by: github-actions <41898282+github-actions[bot]@users.noreply.github.com>

aspicer committed:
feat: move query performance polling to its own celery task in a performant manner (#22497)

mariusandra committed:
feat(hogql): inline filters into subqueries (#22468)

mariusandra committed:
feat(hogql): ignore future persons (#22507)

feedanal committed:
fix: add persistence to redis and kafka on hobby (#22563)

* add persistence to redis and kafka on hobby

This is to address https://github.com/PostHog/posthog/issues/22291, but potentially fixes other redis/kafka related conditions over containers updates / rebuilds

* proper path to kafka data volume

* add volume definitions

---------

Co-authored-by: Frank Hamand <frank@posthog.com>

thmsobrmlr committed:
chore(vscode): adapt celery configuration (#22606)

benjackwhite committed:
fix: Missing scope on events endpoint (#22604)

tomasfarias committed:
revert: Make port a tel (#22609)

zlwaterfield committed:
chore: add ids to admin for qol when debugging/doing support (#22607)

add ids to admin for qol when debugging/doing support

raquelmsmith committed:
feat: add person profiles instructions to SDKs (#22403)

Co-authored-by: github-actions <41898282+github-actions[bot]@users.noreply.github.com>

aspicer committed:
fix(insights): fix stacked line chart (#22598)

aspicer committed:
fix: filter width (#22615)

EDsCODE committed:
feat(data-warehouse): data imports pipeline UI (#22553)

* add data warehouse source tables into pipeline ui

* add empty state

* types

* ui

* total rows synced column

* Update UI snapshots for `chromium` (2)

* Update UI snapshots for `chromium` (1)

* Update UI snapshots for `chromium` (2)

* Update UI snapshots for `chromium` (1)

* add tooltip

---------

Co-authored-by: github-actions <41898282+github-actions[bot]@users.noreply.github.com>

frankh committed:
remove unnecessary deleteRecordSuccess call in managed proxy (#22475)

this was just to restart polling back in the dark ages before we always
ran the poll check

Gilbert09 committed:
fix(data-warehouse): Dont pass password if its not set (#22619)

Dont pass password if its not set

posthog-bot committed:
chore(deps): Update posthog-js to 1.136.4 (#22630)

Gilbert09 committed:
fix(data-warehouse): Fix getting ssh enabled var (#22629)

Fix getting ssh enabled var

pauldambra committed:
chore: add a data attr (#22621)

Gilbert09 committed:
feat(hogql): type system (#22587)

* restore wip

* add numeric operator signatures

* Fix all the unit tests

* Improvements to type system

* Updatd mypy

* Fixed mypy issues

* Fixed property types

* Fixed mypy issues

---------

Co-authored-by: eric <eeoneric@gmail.com>

Gilbert09 committed:
feat(data-warehouse): Resolve expression fields underlying type in db schema query (#22611)

* restore wip

* add numeric operator signatures

* Fix all the unit tests

* Improvements to type system

* Updatd mypy

* Fixed mypy issues

* Fixed property types

* Fixed mypy issues

* Resolve expression fields to their underlying types during the database schema query

---------

Co-authored-by: eric <eeoneric@gmail.com>

robbie-c committed:
fix(web-analytics): Fix searching for session properties with multiple words in the search term (#22632)

Twixes committed:
chore: Add `query-async` to `PERSISTED_FEATURE_FLAGS` (#22628)

Twixes committed:
chore(insights): Consolidate HogQL flags to `hogql-insights-preview` (#22631)

neilkakkar committed:
fix(flags): Better code snippets (#22589)

Commit:
feat: Final piece of deprecating available features, DB references (#22402)

* remove db ref

* update the migration

* remove referencess from test fixture jsons

* some more removals

* Update UI snapshots for `chromium` (1)

* Update UI snapshots for `chromium` (2)

* Update query snapshots

* Update query snapshots

* Update query snapshots

* Update query snapshots

* Update UI snapshots for `chromium` (1)

* Update query snapshots

* fix migration

* add reverse sql

* fix the migration

* Update UI snapshots for `chromium` (2)

* Update query snapshots

* Update UI snapshots for `chromium` (2)

* Update UI snapshots for `chromium` (2)

* Update UI snapshots for `chromium` (1)

* Update UI snapshots for `chromium` (2)

* Update query snapshots

* Update UI snapshots for `chromium` (1)

* Update UI snapshots for `chromium` (2)

* Update UI snapshots for `chromium` (2)

* Update query snapshots

* Update UI snapshots for `chromium` (2)

* Update UI snapshots for `chromium` (2)

* Update UI snapshots for `chromium` (2)

* Update query snapshots

* Update UI snapshots for `chromium` (2)

* Update query snapshots

* Update UI snapshots for `chromium` (2)

* Update query snapshots

* Update query snapshots

* i think i finally fixed the migration issue

* Update UI snapshots for `chromium` (2)

* Update query snapshots

* Update UI snapshots for `chromium` (2)

* Update UI snapshots for `chromium` (2)

* update migration post merge

---------

Co-authored-by: github-actions <41898282+github-actions[bot]@users.noreply.github.com>

Phanatic committed:
Merge branch 'master' into repetable-surveys

github-actions[bot] committed:
Update UI snapshots for `chromium` (1)

github-actions[bot] committed:
Update UI snapshots for `chromium` (1)

github-actions[bot] committed:
Update UI snapshots for `chromium` (2)

Commit:
Add feature flag to UI

Phanatic committed:
Merge branch 'master' into repetable-surveys

github-actions[bot] committed:
Update UI snapshots for `chromium` (2)

Commit:
Merge branch 'master' into repetable-surveys

Commit:
fix feature flag check and billing feature

github-actions[bot] committed:
Update query snapshots

Phanatic committed:
Merge branch 'master' into repetable-surveys

github-actions[bot] committed:
Update query snapshots

github-actions[bot] committed:
Update UI snapshots for `chromium` (2)

Phanatic commented:
### To try this feature out locally. 

1. Create a new feature flag called `surveys-recurring` and enable it to 100% of users
2. You should now see new options available under the `Completion conditions` tab.
3. Create a survey with multiple iterations over a period of days of your choosing.
4. We pick the iterations based on the date, to simulate a later iteration, run this query from a local psql client
``` bash
# password is posthog
psql -h 0.0.0.0 -p 5432 -U posthog

> update posthog_survey set current_iteration=2 where id=<your survey id>
```
5. Now refresh the client-side application, your playground nextjs application, to record a survey response with a later iteration.

Phanatic committed:
Merge branch 'master' into repetable-surveys

github-actions[bot] committed:
Update query snapshots

github-actions[bot] committed:
Update UI snapshots for `chromium` (2)

github-actions[bot] committed:
Update query snapshots

github-actions[bot] committed:
Update UI snapshots for `chromium` (2)

github-actions[bot] committed:
Update query snapshots

github-actions[bot] committed:
Update UI snapshots for `chromium` (2)

github-actions[bot] committed:
Update query snapshots

github-actions[bot] committed:
Update UI snapshots for `chromium` (2)

github-actions[bot] committed:
Update query snapshots

github-actions[bot] committed:
Update UI snapshots for `chromium` (2)

github-actions[bot] committed:
Update query snapshots

github-actions[bot] committed:
Update UI snapshots for `chromium` (2)

raquelmsmith committed:
fix: remove feature gate for action description (#22687)

Co-authored-by: github-actions <41898282+github-actions[bot]@users.noreply.github.com>

github-actions[bot] committed:
Update query snapshots

Commit:
remove test snapshot that's flapping

Phanatic committed:
Merge branch 'master' into repetable-surveys

github-actions[bot] committed:
Update UI snapshots for `chromium` (2)

github-actions[bot] committed:
Update UI snapshots for `chromium` (1)

github-actions[bot] committed:
Update UI snapshots for `chromium` (2)

github-actions[bot] committed:
Update UI snapshots for `chromium` (1)

github-actions[bot] committed:
Update UI snapshots for `chromium` (2)

Phanatic committed:
Merge branch 'master' into repetable-surveys

Phanatic committed:
Merge branch 'master' into repetable-surveys

github-actions[bot] committed:
Update UI snapshots for `chromium` (2)

github-actions[bot] committed:
Update UI snapshots for `chromium` (1)

github-actions[bot] committed:
Update UI snapshots for `chromium` (2)

github-actions[bot] committed:
Update UI snapshots for `chromium` (1)

github-actions[bot] committed:
Update UI snapshots for `chromium` (2)

github-actions[bot] committed:
Update UI snapshots for `chromium` (2)

github-actions[bot] committed:
Update UI snapshots for `chromium` (1)

github-actions[bot] committed:
Update UI snapshots for `chromium` (2)

Phanatic committed:
Merge branch 'master' into repetable-surveys

github-actions[bot] committed:
Update UI snapshots for `chromium` (1)

github-actions[bot] committed:
Update UI snapshots for `chromium` (2)

github-actions[bot] committed:
Update UI snapshots for `chromium` (2)

Commit:
Merge branch 'master' into repetable-surveys

Commit:
Merge branch 'master' into repetable-surveys

jurajmajerik commented on frontend/src/scenes/surveys/SurveyView.tsx:
```diff
@@ -307,17 +336,34 @@ export function SurveyResult({ disableEventsTable }: { disableEventsTable?: bool
+                                    iteration={survey.current_iteration}
                                 />
+
+                                {survey.iteration_count &&
+                                    survey.iteration_count > 0 &&
+                                    survey.current_iteration &&
+                                    survey.current_iteration > 0 &&
+                                    survey.iteration_start_dates &&
+                                    survey.iteration_start_dates.length > 0 && (
```
nit: consider abstracting this into a variable to keep the markdown clean

jurajmajerik reviewed (approved):
LGTM, assuming Neil's comments are addressed!

General feedback: I think this could well be split into two PRs (UI + backend) for easier and quicker review :)

Phanatic committed:
Merge branch 'master' into repetable-surveys

jurajmajerik commented:
One more thing - this deserves its own section in the docs. I'd suggest putting it within Surveys/Features, like survey webhooks: https://posthog.com/docs/surveys/webhook

We typically add to the docs ourselves and request a review from Ian or Lior.

Phanatic committed:
Merge branch 'master' into repetable-surveys

zlwaterfield committed:
feat: better proration calculations (#22717)

* make proration calcs better

* improve proration var naming

* improve prorated copy

* Update UI snapshots for `chromium` (2)

---------

Co-authored-by: github-actions <41898282+github-actions[bot]@users.noreply.github.com>

jurajmajerik commented:
Something else I've noticed after linking posthog-js - the Query doesn't seem to be showing a correct result:

![image](https://github.com/PostHog/posthog/assets/22996112/fec34441-121e-47d7-b0b8-87b8c2ca28af)

tiina303 committed:
feat: track  usage outside of known events (#22755)

jurajmajerik commented:
And one design nit - please add some margin here:

![Screenshot 2024-06-06 at 18 19 01](https://github.com/PostHog/posthog/assets/22996112/498112b4-becf-44cf-8746-48ec448f9b92)

daibhin committed:
chore: remove antd Empty component (#22655)

Co-authored-by: github-actions <41898282+github-actions[bot]@users.noreply.github.com>

Phanatic commented:
good eye @jurajmajerik ,  fixed now
<img width="505" alt="image" src="https://github.com/PostHog/posthog/assets/82819/6023900b-3939-4243-8fb1-5863eb00d88e">

zlwaterfield committed:
chore: updating activation endpoint to be activate (#22719)

* /activation -> /activate

* update the billing activate viewset actions

* update billing

xrdt committed:
chore: Move some print statements in usage_reports to logger (#22737)

move some print statements in usage_reports to logger

Commit:
add margin to schedule input:

Commit:
Merge branch 'master' into repetable-surveys

github-actions[bot] committed:
Update UI snapshots for `chromium` (2)

github-actions[bot] committed:
Update UI snapshots for `chromium` (1)

github-actions[bot] committed:
Update UI snapshots for `chromium` (2)

dmarticus reviewed (commented):
some minor grammar comments that I noticed hadn't been cleaned up yet, going to do a more robust review after I've eaten lunch,.

github-actions[bot] committed:
Update UI snapshots for `chromium` (1)

github-actions[bot] committed:
Update UI snapshots for `chromium` (2)

zlwaterfield committed:
feat: better billing limits UI (#22726)

* improve billing limit feature

* Update BillingLimitInput.tsx

* Update UI snapshots for `chromium` (1)

* billing limit improvements

* Update BillingLimit.tsx

* improve copy

* Update frontend/src/scenes/billing/BillingLimit.tsx

Co-authored-by: Bianca Yang <21014901+xrdt@users.noreply.github.com>

* Update frontend/src/scenes/billing/BillingLimit.tsx

* Update frontend/src/scenes/billing/BillingLimit.tsx

* Update UI snapshots for `chromium` (1)

* Update UI snapshots for `chromium` (2)

* prettier

---------

Co-authored-by: github-actions <41898282+github-actions[bot]@users.noreply.github.com>
Co-authored-by: Bianca Yang <21014901+xrdt@users.noreply.github.com>

tiina303 committed:
fix: Move set usage tracking earlier (#22761)

github-actions[bot] committed:
Update UI snapshots for `chromium` (2)

Commit:
fix some text

Commit:
Merge branch 'master' into repetable-surveys

Commit:
Merge branch 'master' into repetable-surveys

github-actions[bot] committed:
Update UI snapshots for `chromium` (2)

github-actions[bot] committed:
Update UI snapshots for `chromium` (2)

Commit:
Add vertical spacing

Commit:
Merge branch 'master' into repetable-surveys

github-actions[bot] committed:
Update UI snapshots for `chromium` (1)

github-actions[bot] committed:
Update UI snapshots for `chromium` (2)

github-actions[bot] committed:
Update UI snapshots for `chromium` (1)

zlwaterfield committed:
chore: updating /activation endpoint to /activate (#22779)

* Move /activation -> /activate

* Update UI snapshots for `chromium` (2)

* Update UI snapshots for `chromium` (2)

* Update UI snapshots for `chromium` (2)

---------

Co-authored-by: github-actions <41898282+github-actions[bot]@users.noreply.github.com>

Commit:
fix tests

github-actions[bot] committed:
Update UI snapshots for `chromium` (1)

github-actions[bot] committed:
Update UI snapshots for `chromium` (2)

github-actions[bot] committed:
Update UI snapshots for `chromium` (1)

github-actions[bot] committed:
Update UI snapshots for `chromium` (2)

github-actions[bot] committed:
Update UI snapshots for `chromium` (2)

github-actions[bot] committed:
Update UI snapshots for `chromium` (2)

github-actions[bot] committed:
Update UI snapshots for `chromium` (2)

raquelmsmith committed:
fix: remove negation operator from check for disabledReason (#22782)

Co-authored-by: github-actions <41898282+github-actions[bot]@users.noreply.github.com>

pauldambra committed:
chore: metric for batch insight caching state updates (#22776)

webjunkie committed:
feat(insights): Async calculation for saved insights/dashboards (#22586)

tomasfarias committed:
feat: Timestamp-based backfills for batch exports (#22673)

* refactor: Backfills now use start_workflow instead of backfill API

* feat: Support is_backfill parameter to use timestamp in queries

* feat: Pass is_backfill on backfill runs for all batch exports

* fix: Add type hints

* fix: Add new inputs to test

* refactor: Attempt cancelling backfill before deleting schedule

* fix: Use timezone aware dates

* feat: Cast to team's timezone if available

* fix: Do not account for jitter anymore

* fix: Use right ID in test

pauldambra committed:
feat: move session replay pinned persistence to its own queue (#22765)

pauldambra committed:
fix: heatmap only in-app not click-map (#22786)

Co-authored-by: github-actions <41898282+github-actions[bot]@users.noreply.github.com>

pauldambra committed:
fix: heatmap in-app browser on small screen (#22788)

Co-authored-by: github-actions <41898282+github-actions[bot]@users.noreply.github.com>

pauldambra committed:
fix: use keyboard navigation in toolbar (#22792)

mariusandra committed:
feat(hog): hoge tests, cli and debugger (#22743)

webjunkie committed:
fix(insights): Change refresh modes for dashboard to blocking (#22799)

Change refresh modes for dashboard to blocking

benjackwhite committed:
chore: Export Hog interfaces (#22798)

EDsCODE committed:
fix(data-warehouse): fix view schema bug (#22763)

* fix view schema bug

* Update UI snapshots for `chromium` (2)

* Update UI snapshots for `chromium` (2)

* Update UI snapshots for `chromium` (1)

* Update UI snapshots for `chromium` (2)

* Update UI snapshots for `chromium` (1)

* Update UI snapshots for `chromium` (2)

* Update UI snapshots for `chromium` (2)

---------

Co-authored-by: github-actions <41898282+github-actions[bot]@users.noreply.github.com>

pauldambra committed:
feat: collapsible setting panel for in-app heatmaps (#22801)

Co-authored-by: github-actions <41898282+github-actions[bot]@users.noreply.github.com>

pauldambra committed:
fix: allow x-highlight-request tracing header (#22805)

Commit:
fix tests

Phanatic committed:
Merge branch 'master' into repetable-surveys

github-actions[bot] committed:
Update UI snapshots for `chromium` (2)
